### DISCUSSION TITLE
Problem displaying upper case umlauts in foot

### DISCUSSION THREAD

**Discussion by heapifyman** [Manjaro-Sway/manjaro-sway#459] [Q&A] [answered]
Is anyone else having problems with displaying upper case umlauts (Ö, Ä, Ü) in foot? For me it looks like they are cut off at the top, the dots are not or barely visible.
It's not a big issue but sometimes there are filenames that contain upper case umlauts.

Using default `foot.ini`.

On Laptop screen (1920x1080):
![umlaut-1920x1080](https://user-images.githubusercontent.com/274236/220860925-a0eb5bb2-d772-46cc-8c41-ade965ff8366.png)

On external screen (2560x1440):
![umlaut-2560x1440](https://user-images.githubusercontent.com/274236/220860950-680191bb-15e6-4008-bbd8-cee8b790e9e0.png)

I also notice that `[` and `]` (maybe other characters as well) seem to be cut off at the top in neovim.

![swappy-20230223_095216](https://user-images.githubusercontent.com/274236/220861057-cdcaccb6-7d2d-42b0-b84e-08ecc0160ee0.png)

**Comment by boredland** [comment]
seems to depend on the resolution then. at least on my ~3k screen it looks fine:

![image](https://user-images.githubusercontent.com/4662748/220885006-291f5c7b-2cbc-40fe-b869-cd0826b74292.png)

Havent had a fhd screen for quite some time. it looks like the line height doesn't match the height of the chars.

**Reply by heapifyman** [reply]
Any suggestions? I tried changing `line-height` or `vertical-letter-offset`. `vertical-letter-offset` seems to help but breaks the prompt (powerlevel10k ?). Btw, I haven't changed any display settings, scale factor is 1.

![swappy-20230225_162926](https://user-images.githubusercontent.com/274236/221365466-4f8c923b-2f32-4a9d-8eda-58efe7d35a82.png)

> Havent had a fhd screen for quite some time.

I think FHD displays are still very common, especially on "business laptops". For example, the majority of Dell Laptops seem to have a FHD resolution.

**Reply by boredland** [reply]
I think I have a FHD laptop here. But I won't do something really different  than tinkering with the foot settings. Does a smaller font size give you better results?

**Reply by heapifyman** [reply]
Default font size `font=RobotoMono Nerd Font Mono:size=8` is pretty small already.

Setting `font=RobotoMono Nerd Font Mono:size=6` seems to improve it. But at my age that is way too small to read :(

![size6](https://user-images.githubusercontent.com/274236/221374342-3e72973c-6924-42d8-8019-72faaead1130.png)

Actually, I would prefer setting font size to 10.

**Reply by heapifyman** [reply, marked as answer]
`/usr/share/zsh-theme-powerlevel10k/font.md` recommends `Meslo Nerd Font patched for Powerlevel10k`.

So I tried installing `ttf-meslo-nerd-font-powerlevel10k` and setting `font=MesloLGS NF:size=8` in `~/.config/foot/foot.ini`. And that seems to have helped.

**Reply by boredland** [reply]
I would accept a PR switching from RobotoMono to Meslo, but to myself it isn't a priority tbh.

**Reply by boredland** [reply]
in `manjaro-sway-settings-git`, we now use the meslo font - feel free to give it a try.

**Comment by ishaanbhimwal** [comment]
@boredland not important but can you give jetbrains mono a try. It's a great monospace font and suits Manjaro sway really well.

```
sudo pacman -S ttf-jetbrains-mono-nerd
```

![image](https://user-images.githubusercontent.com/79986754/235856543-49e86c22-615e-4716-b1a7-c4c53e313584.png)

**Reply by boredland** [reply]
Nice, will give that a try. So far I don't really like the meslo font.

**Reply by heapifyman** [reply]
JetBrains Mono looks better 👍🏽 .

Only the icons in the prompt are really small. Is there a way to increase their size?

**Reply by ishaanbhimwal** [reply]
Nice that you liked the font. It's free and open source and one of the favorites of r/unixporn. Also resembles the boring Roboto Mono a tiny bit and look cool at the same time. [Website](https://www.jetbrains.com/lp/mono/)

The icons should not look small. Make sure you are not using the normal Jetbrains Mono font and using the Nerd version "JetBrainsMono Nerd Font" installed via `ttf-jetbrains-mono-nerd`.

**Reply by heapifyman** [reply]
Had the right package installed but a typo in my `foot.ini`. Apologies.

**Reply by boredland** [reply]
I really like jetbrains mono. I'd propose that we replace roboto with it, but leave roboto installed for the time being, so stuff doesn't completely break for lots of people.

**Reply by boredland** [reply]
sooo, `manjaro-sway-settings-git` should only contain jetbrains mono now. would love for some ppl to try it tho.

**Reply by ishaanbhimwal** [reply]
If you want, I or you can make a post in r/manjaro, asking people for suggestions about this change.

**Reply by boredland** [reply]
sure, but lets make sure the package is in the repos first.

**Reply by boredland** [reply]
package should be in the repo - make sure you run `skel` after installing, as many userspace configurations needed to be adapted.

**Reply by boredland** [reply]
Let me know when you posted. My first impression is that it fits quite nicely for vscode, waybar and foot - the tools I use the most 🖥️

**Reply by ishaanbhimwal** [reply]
[Posted here](https://www.reddit.com/r/ManjaroLinux/comments/138u4tl/manjaro_sway_users_we_need_your_input/). It's always better to take input from actual users before implementing such changes.

**Reply by boredland** [reply]
hm, my battery icon is a sadface emoji:

![image](https://user-images.githubusercontent.com/4662748/236525015-a7a04ecc-3aa9-4426-8d5a-ff808caf6a96.png)

and the wlsunset active icon looks like the airbnb log:

![image](https://user-images.githubusercontent.com/4662748/236525274-195c535d-6692-48cd-bc7d-583bb8f76105.png)

**Reply by boredland** [reply]
seems nerdfonts removed the "-mdi" variants that I used in lots of places. damn, that is annoying.

**Reply by boredland** [reply]
Well, I guess we should release as a major and fix along the way. I used it for some days extensively, no more glitches spotted.

**Reply by boredland** [reply]
Well, I guess we should release as a major and fix along the way. I used it for some days extensively, no more glitches spotted.

**Reply by ishaanbhimwal** [reply]
The new fonts added [here](https://github.com/manjaro-sway/desktop-settings/commit/64000e6d7f857be20e5197ac6b26716dc3a7234a) look really large compared to the old ones. Although this is probably due to the nerdfonts update, is there any way to fix this? Example, see the battery, bluetooth and wlsunset icon in the image below:

![swappy-20230512_012656](https://github.com/manjaro-sway/manjaro-sway/assets/79986754/49a7f99c-e9ec-4d5e-9840-39b92a15e7cb)

**Reply by boredland** [reply]
I guess we should be able to align them using the style.css for waybar - PR welcome ;-)

**Reply by ishaanbhimwal** [reply]
how will this update reach the end users? I mean when will the package hit unstable repos?

**Reply by boredland** [reply]
I won't, our packages are released independently. You can already install `manjaro-sway-settings-git`. And run "skel" afterwards. Looking at your icons, you may not have done that?

**Reply by boredland** [reply]
Had another look, it looks very similar to yours. Especially the too big Bluetooth button is annoying.
